### PR TITLE
PMA-009

### PR DESCRIPTION
**Pull request by AlexanderSUS**
1. Create Login and registration pages
2. Create authorization services
3. Create authorization state
4. Create log in, sign up, log out buttons

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable no-param-reassign */
```
Comment (allyksander) [inline src/api/index.ts]: May be better change rule for eslint?

File: src/components/LoginForm/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
Comment (allyksander) [inline src/components/LoginForm/LoginForm.tsx]: Same as above

File: src/components/RegistrationForm/RegistrationFrom.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
Comment (allyksander) [inline src/components/RegistrationForm/RegistrationFrom.tsx]: Same as above

File: src/store/authSlice.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
+/* eslint-disable import/no-cycle */
+/* eslint-disable no-param-reassign */
```
Comment (allyksander) [inline src/store/authSlice.ts]: Same as above

**Review by allyksander — APPROVED** [review]

**Comment by allyksander** [comment]
Also I have this problem after start project https://prnt.sc/-_E1CGQ1i2XD

File: src/store/authSlice.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { AxiosError } from 'axios';
+import jwt_decode from 'jwt-decode';
+import AuthService from '../services/authService';
+import {
+  SignUpResponse, SignInResponse, ValidationErrors, ErrorResponseData,
+} from '../types/response';
+import { NewUser, User } from '../types/user';
+import { RootState } from './store';
```
Comment (Gibr) [inline src/store/authSlice.ts]: we can avoid using /* eslint-disable import/no-cycle */ here by changing this string to:
import type { RootState } from './store';

File: src/api/index.ts (hunk 1 above)
Comment (AlexanderSUS) [reply src/api/index.ts]: fix esllint rule to warning instead error

File: src/components/LoginForm/LoginForm.tsx (hunk 2 above)
Comment (AlexanderSUS) [reply src/components/LoginForm/LoginForm.tsx]: fix esllint rule to warning instead error

File: src/components/RegistrationForm/RegistrationFrom.tsx (hunk 3 above)
Comment (AlexanderSUS) [reply src/components/RegistrationForm/RegistrationFrom.tsx]: fix esllint rule to warning instead error

File: src/store/authSlice.ts (hunk 4 above)
Comment (AlexanderSUS) [reply src/store/authSlice.ts]: fix esllint rule to warning instead error

File: src/store/authSlice.ts (hunk 5 above)
Comment (AlexanderSUS) [reply src/store/authSlice.ts]: Done, didn't know about this option

**Pull request merged**